### PR TITLE
[SRVKS-1262] ContainerHealthy reset

### PR DESCRIPTION
**Pull request by skonto**
**What this PR does / why we need it**:

- Temporary fix suggested upstream. 
- See related jira for the details.

### CODE REVIEW

File: pkg/reconciler/revision/reconcile_resources.go [hunk 1]
```diff
@@ -117,6 +117,9 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1.Revision)
 		}
 	}
 
+	if *deployment.Spec.Replicas > 0 && *deployment.Spec.Replicas == deployment.Status.ReadyReplicas {
```
Comment (maschmid) [inline pkg/reconciler/revision/reconcile_resources.go:120]: Should it be compared to AvailableReplicas instead?

Comment (skonto) [reply pkg/reconciler/revision/reconcile_resources.go:120]: Ready means available as we the deployment we create has the following set to 0:
```
	// Minimum number of seconds for which a newly created pod should be ready
	// without any of its container crashing, for it to be considered available.
	// Defaults to 0 (pod will be considered available as soon as it is ready)
	// +optional
	MinReadySeconds int32 `json:"minReadySeconds,omitempty" protobuf:"varint,5,opt,name=minReadySeconds"`
```

**Pull request merged**